bin/desi_tile_redshifts

```diff
@@ -0,0 +1,343 @@
+#!/usr/bin/env python
+
+"""
+Another version of running redshifts per tile
+
+Examples:
+
+All exposures of tile 80605 on night 20201215:
+
+    desi_tile_redshifts --tile 80605 --night 20201215
+
+Exposures E1 E2 E3 on night 20201215 (auto splitting by TILEID if needed):
+
+    desi_tile_redshifts --night 20201215 --expid 67972 67973 67968 67969
+
+Tile 80605 combined across multiple nights:
+
+    desi_tile_redshifts --tile 80605 --night 20201214 20201215 --group blat
+
+Tile 80605 combined across all nights:
+
+    desi_tile_redshifts --tile 80605 --group all
+
+Generate scripts for every tile on 20201215 but don't submit batch jobs:
+
+    desi_tile_redshifts --night 20201215 --nosubmit
+
+Use exposures from a separately curated input list:
+
+    desi_tile_redshifts --explist explist-deep.txt --group deep
+
+Not supported yet: multiple tiles on a single night in a single call:
+
+    desi_tile_redshifts --night 20201215 --tileid 80605 80606 80607
+
+"""
+
+#- Parse command line quickly for --help before slower imports
+import argparse
+p = argparse.ArgumentParser()
+p.add_argument("-n", "--night", type=int, nargs='+', help="YEARMMDD nights")
+p.add_argument("-t", "--tileid", type=int, help="Tile ID")
+p.add_argument("-e", "--expid", type=int, nargs='+', help="exposure IDs")
+p.add_argument("-g", "--group", type=str, help="night group name")
+p.add_argument("--explist", type=str,
+        help="read TILE NIGHT EXPID list from this file")
+p.add_argument("--nosubmit", action="store_true",
+        help="generate scripts but don't submit batch jobs")
+p.add_argument("--batch-queue", type=str, default='realtime',
+        help="batch queue name")
+p.add_argument("--batch-reservation", type=str,
+        help="batch reservation name")
+p.add_argument("--batch-dependency", type=str,
+        help="job dependencies passed to sbatch --dependency")
+
+args = p.parse_args()
+
+import sys, os, glob
+import subprocess
+import numpy as np
+from astropy.table import Table, vstack
+
+from desiutil.log import get_logger
+
+import desispec.io
+from desispec.workflow.tableio import load_table
+from desispec.workflow.exptable import get_exposure_table_pathname
+
+def batch_tile_redshifts(tileid, exptable, group, spectrographs=None,
+    submit=False, queue='realtime', reservation=None, dependency=None):
+    """
+    Generate batch script for spectra+coadd+redshifts for a tile
+
+    Args:
+        tileid (int): Tile ID
+        exptable (Table): has columns NIGHT EXPID to include
```

Comment:
I added comments to clarify that.  This was partially motivated by the change of exposure_tables format during sv1 which tripped up stacking them across all nights.  Trimming down to just the columns needed and having the functions only require the minimal set of columns needed (as a generic Table) instead of the specific pipeline full exposures_table made this more robust to the exact input format (other columns can be there; they are just ignored)